### ISSUE TITLE
AssertionError: found 1 duplicate font objs in GS output

### ISSUE THREAD

**Issue by rbrito** [pts/pdfsizeopt#49]
Dear @pts,

I just discovered another file here that I processed (also) with Acrobat Pro 8 in the past and pdfsizeopt has problems with it:

```
info: This is pdfsizeopt rUNKNOWN size=378552.
info: prepending to PATH: /home/rbrito/Downloads/pdfsizeopt
info: loading PDF from: 10.1.1.121.6177.opt.pdf
info: loaded PDF of 78920 bytes
info: using Ghostscript /usr/bin/gs: GPL Ghostscript 9.21 (2017-03-16)
info: decompressing 48 bytes with Ghostscript /Filter/FlateDecode/DecodeParms <</Columns 4/Predictor 12>>
info: decompressing 170 bytes with Ghostscript /Filter/FlateDecode/DecodeParms <</Columns 5/Predictor 12>>
info: found 101 obj offsets and 4 obj streams in xref stream
info: separated to 95 objs + xref + trailer
info: parsed 95 objs
info: found 2 Type1 fonts loaded
info: writing Type1CConverter (6859 font bytes) to: psotmp.22773.conv.tmp.ps
info: executing Type1CConverter with Ghostscript: gs -q -P- -dNOPAUSE -dBATCH -sDEVICE=pdfwrite -dPDFSETTINGS=/printer -dColorConversionStrategy=/LeaveColorUnchanged -sOutputFile=psotmp.22773.conv.tmp.pdf -f psotmp.22773.conv.tmp.ps
Type1CConverter: using interpreter GPL Ghostscript 921 20170316
Type1CConverter: converting font /FKBKGI+mtsy to /Obj0000000057
warning: using glyphshow for unencoded glyph: /.notdef
Type1CConverter: converting font /FKBKGI+rmtmi to /Obj0000000061
warning: using glyphshow for unencoded glyph: /.notdef
Type1CConverter: all OK
info: loading PDF from: psotmp.22773.conv.tmp.pdf
info: loaded PDF of 6523 bytes
info: separated to 23 objs + xref + trailer
info: parsed 23 objs
error: duplicate font /Obj0000000057 obj old=57 new=17
info: found 3 fonts in GS output
Traceback (most recent call last):
  File "/home/rbrito/Downloads/pdfsizeopt/pdfsizeopt", line 41, in <module>
    sys.exit(main.main(sys.argv, script_dir=script_dir))
  File "/home/rbrito/Downloads/pdfsizeopt/lib/pdfsizeopt/main.py", line 9092, in main
    pdf.ConvertType1FontsToType1C()
  File "/home/rbrito/Downloads/pdfsizeopt/lib/pdfsizeopt/main.py", line 5626, in ConvertType1FontsToType1C
    TMP_PREFIX + 'conv.tmp.ps', TMP_PREFIX + 'conv.tmp.pdf')
  File "/home/rbrito/Downloads/pdfsizeopt/lib/pdfsizeopt/main.py", line 5452, in GenerateType1CFontsFromType1
    do_obj_num_from_font_name=True, where='in GS output')
  File "/home/rbrito/Downloads/pdfsizeopt/lib/pdfsizeopt/main.py", line 5367, in GetFonts
    'found %d duplicate font objs %s' % (duplicate_count, where))
AssertionError: found 1 duplicate font objs in GS output
```

I'm attaching here the corresponding file:

[10.1.1.121.6177.opt.pdf](https://github.com/pts/pdfsizeopt/files/1350875/10.1.1.121.6177.opt.pdf)

Oh, just like the previous issue, which I filed a few minutes ago, the problem can be worked-around if I specify `--do-optimize-fonts=no`.

Thanks,

Rogério.

**Comment by pts** [comment]
Thank you for reporting this!

Based on a short investigation, this is a bug in Type1CConverter creating `/Encoding` incorrectly. The correct behavior would be adding all glyph names in `/CharStrings` to `/Encoding`, provided that there are at most 256 glyphs in `/CharStrings`.

**Comment by pts** [comment]
Fixed it in 26ef82d41348204d93a5421c7750c0ee6df572b5.

**Comment by rbrito** [comment]
Thanks, this worked.
